api/types.go

```diff
@@ -698,7 +698,10 @@ func (pin *Pin) Equals(pin2 *Pin) bool {
 	}
 
 	if pin == pin2 {
-		return true
+		// ask @lanzafame why this is not true
+		// in any case, this is anomalous and we should
+		// not be using this with two nils.
+		return false
```

Comment:
It is `true`
https://play.golang.org/p/yyDohd_hDPk

Reply:
we are conveying some meaning in saying that two `nil` pins are not equal when nil because, for a start, you have nothing to compare. At least I think that was @lanzafame's point.